### ISSUE TITLE
WandbLogger initialization results in Segmentation fault

### ISSUE THREAD

**Issue by DucoG** [Lightning-AI/pytorch-lightning#19161]
### Bug description

simply initialization of a `WandbLogger` instance results in Segmentation fault. This happens in any context. 

here a very basic example:
```python
from lightning.pytorch.loggers import WandbLogger
import time

logger = WandbLogger()

# wait for logger to be initialized
while not isinstance(logger.experiment.name, str):
    print('waiting for logger to be initialized...')
    # wait .1 seconds
    time.sleep(.1)
    continue
```

results in
```
Segmentation fault (core dumped)
```

### What version are you seeing the problem on?

master

### How to reproduce the bug

```python
from lightning.pytorch.loggers import WandbLogger
import time

logger = WandbLogger()

# wait for logger to be initialized
while not isinstance(logger.experiment.name, str):
    print('waiting for logger to be initialized...')
    # wait .1 seconds
    time.sleep(.1)
    continue
```

### Error messages and logs

```
Segmentation fault (core dumped)
```

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
                - NVIDIA GeForce RTX 2080 Ti
        - available:         True
        - version:           11.7
* Lightning:
        - lightning:         2.1.2
        - lightning-cloud:   0.5.39
        - lightning-utilities: 0.10.0
        - pytorch-lightning: 2.1.2
        - pytorch-transformers: 1.2.0
        - torch:             1.13.1
        - torchaudio:        0.13.1
        - torchmetrics:      1.2.1
        - torchsummary:      1.5.1
        - torchvision:       0.14.1
* Packages:
        - absl-py:           2.0.0
        - accelerate:        0.23.0
        - aiofiles:          22.1.0
        - aiohttp:           3.8.5
        - aiosignal:         1.3.1
        - aiosqlite:         0.19.0
        - annotated-types:   0.6.0
        - antlr4-python3-runtime: 4.9.3
        - anyio:             3.7.1
        - aplus:             0.11.0
        - appdirs:           1.4.4
        - argon2-cffi:       23.1.0
        - argon2-cffi-bindings: 21.2.0
        - argparse:          1.4.0
        - arrow:             1.3.0
        - astropy:           6.0.0
        - astropy-iers-data: 0.2023.12.11.0.31.11
        - asttokens:         2.4.1
        - async-timeout:     4.0.3
        - attrs:             23.1.0
        - babel:             2.14.0
        - backoff:           2.2.1
        - batchgenerators:   0.24
        - beautifulsoup4:    4.12.2
        - bitsandbytes:      0.41.2.post2
        - blake3:            0.3.3
        - bleach:            6.1.0
        - blessed:           1.19.1
        - bokeh:             3.3.2
        - boto3:             1.28.57
        - botocore:          1.31.57
        - bottleneck:        1.3.7
        - bqplot:            0.12.42
        - branca:            0.7.0
        - brotli:            1.0.9
        - build:             1.0.3
        - cachecontrol:      0.13.1
        - cached-property:   1.5.2
        - cachetools:        5.3.1
        - certifi:           2023.11.17
        - cffi:              1.16.0
        - charset-normalizer: 3.3.2
        - cleo:              2.1.0
        - click:             8.1.7
        - cloudpickle:       3.0.0
        - colorama:          0.4.6
        - coloredlogs:       15.0.1
        - comm:              0.1.4
        - contourpy:         1.2.0
        - cramjam:           2.7.0
        - crashtest:         0.4.1
        - croniter:          1.4.1
        - cryptography:      41.0.7
        - cycler:            0.12.1
        - cytoolz:           0.12.2
        - dask:              2023.12.0
        - dask-jobqueue:     0.8.2
        - datasets:          2.14.5
        - dateutils:         0.6.12
        - debugpy:           1.8.0
        - decorator:         5.1.1
        - deepdiff:          6.7.1
        - defusedxml:        0.7.1
        - dicom2nifti:       2.4.7
        - dill:              0.3.7
        - distlib:           0.3.8
        - distributed:       2023.12.0
        - docker-pycreds:    0.4.0
        - docstring-parser:  0.15
        - dulwich:           0.21.7
        - einops:            0.6.1
        - entrypoints:       0.4
        - exceptiongroup:    1.2.0
        - executing:         2.0.1
        - fastapi:           0.105.0
        - fastcore:          1.5.27
        - fastjsonschema:    2.19.0
        - fastparquet:       2023.4.0
        - filelock:          3.13.1
        - fonttools:         4.46.0
        - fqdn:              1.5.1
        - frozendict:        2.3.10
        - frozenlist:        1.4.0
        - fsspec:            2023.12.2
        - future:            0.18.2
        - gitdb:             4.0.10
        - gitpython:         3.1.37
        - google-auth:       2.23.3
        - google-auth-oauthlib: 1.0.0
        - graphviz:          0.20.1
        - grpcio:            1.59.0
        - h11:               0.14.0
        - h5py:              3.7.0
        - hiddenlayer:       0.3
        - huggingface-hub:   0.16.4
        - humanfriendly:     10.0
        - hydra-core:        1.3.2
        - idna:              3.6
        - imagecodecs:       2023.1.23
        - imageio:           2.23.0
        - importlib-metadata: 7.0.0
        - importlib-resources: 6.1.1
        - inquirer:          3.1.4
        - installer:         0.7.0
        - ipydatawidgets:    4.3.2
        - ipykernel:         6.26.0
        - ipyleaflet:        0.18.0
        - ipympl:            0.9.3
        - ipython:           8.18.1
        - ipython-genutils:  0.2.0
        - ipyvolume:         0.6.3
        - ipyvue:            1.10.1
        - ipyvuetify:        1.8.10
        - ipywebrtc:         0.6.0
        - ipywidgets:        8.1.1
        - isoduration:       20.11.0
        - itsdangerous:      2.1.2
        - jaraco.classes:    3.3.0
        - jedi:              0.19.1
        - jeepney:           0.8.0
        - jinja2:            3.1.2
        - jmespath:          1.0.1
        - joblib:            1.3.2
        - json5:             0.9.14
        - jsonargparse:      4.27.0
        - jsonpointer:       2.4
        - jsonschema:        4.20.0
        - jsonschema-specifications: 2023.11.2
        - jupyter-client:    7.4.9
        - jupyter-core:      5.5.0
        - jupyter-events:    0.9.0
        - jupyter-server:    2.12.1
        - jupyter-server-fileid: 0.9.0
        - jupyter-server-terminals: 0.5.0
        - jupyter-server-ydoc: 0.8.0
        - jupyter-ydoc:      0.2.4
        - jupyterlab:        3.6.3
        - jupyterlab-pygments: 0.3.0
        - jupyterlab-server: 2.25.2
        - jupyterlab-widgets: 3.0.9
        - keyring:           24.3.0
        - kiwisolver:        1.4.5
        - lazy-loader:       0.3
        - lightning:         2.1.2
        - lightning-cloud:   0.5.39
        - lightning-utilities: 0.10.0
        - linecache2:        1.0.0
        - llvmlite:          0.41.1
        - locket:            1.0.0
        - lovely-numpy:      0.2.8
        - lovely-tensors:    0.1.14
        - lz4:               4.3.2
        - markdown:          3.5
        - markdown-it-py:    3.0.0
        - markupsafe:        2.1.3
        - matplotlib:        3.6.2
        - matplotlib-inline: 0.1.6
        - mdurl:             0.1.0
        - medpy:             0.4.0
        - mistune:           3.0.2
        - mkl-fft:           1.3.8
        - mkl-random:        1.2.4
        - mkl-service:       2.4.0
        - more-itertools:    10.1.0
        - mpmath:            1.3.0
        - msgpack:           1.0.7
        - multidict:         6.0.4
        - multiprocess:      0.70.15
        - munkres:           1.1.4
        - nbclassic:         1.0.0
        - nbclient:          0.8.0
        - nbconvert:         7.12.0
        - nbformat:          5.9.2
        - nest-asyncio:      1.5.8
        - networkx:          2.8.8
        - nibabel:           4.0.2
        - nnunet:            1.7.0
        - notebook:          6.5.6
        - notebook-shim:     0.2.3
        - numba:             0.58.1
        - numexpr:           2.8.7
        - numpy:             1.26.2
        - oauthlib:          3.2.2
        - omegaconf:         2.3.0
        - optimum:           1.13.2
        - ordered-set:       4.1.0
        - orjson:            3.9.10
        - overrides:         7.4.0
        - packaging:         23.2
        - pandas:            1.5.2
        - pandocfilters:     1.5.0
        - parso:             0.8.3
        - partd:             1.4.1
        - pathtools:         0.1.2
        - pexpect:           4.8.0
        - pickleshare:       0.7.5
        - pillow:            10.0.1
        - pip:               23.3.1
        - pkginfo:           1.9.6
        - pkgutil-resolve-name: 1.3.10
        - platformdirs:      3.11.0
        - ply:               3.11
        - poetry:            1.7.1
        - poetry-core:       1.8.1
        - poetry-plugin-export: 1.6.0
        - progressbar2:      4.2.0
        - prometheus-client: 0.19.0
        - prompt-toolkit:    3.0.42
        - protobuf:          3.20.2
        - psutil:            5.9.5
        - ptyprocess:        0.7.0
        - pure-eval:         0.2.2
        - py-cpuinfo:        9.0.0
        - pyarrow:           11.0.0
        - pyarrow-hotfix:    0.6
        - pyasn1:            0.5.0
        - pyasn1-modules:    0.3.0
        - pycparser:         2.21
        - pydantic:          2.1.1
        - pydantic-core:     2.4.0
        - pydicom:           2.3.1
        - pyerfa:            2.0.1.1
        - pygments:          2.17.2
        - pyjwt:             2.8.0
        - pynndescent:       0.5.11
        - pyparsing:         3.1.1
        - pyproject-hooks:   1.0.0
        - pyqt5:             5.15.10
        - pyqt5-sip:         12.13.0
        - pysocks:           1.7.1
        - python-dateutil:   2.8.2
        - python-editor:     1.0.4
        - python-gdcm:       3.0.20
        - python-json-logger: 2.0.7
        - python-multipart:  0.0.6
        - python-utils:      3.8.1
        - pythreejs:         2.4.2
        - pytorch-lightning: 2.1.2
        - pytorch-transformers: 1.2.0
        - pytz:              2023.3.post1
        - pywavelets:        1.4.1
        - pyyaml:            6.0.1
        - pyzmq:             24.0.1
        - rapidfuzz:         3.5.2
        - readchar:          4.0.5.dev0
        - referencing:       0.32.0
        - regex:             2023.8.8
        - requests:          2.31.0
        - requests-oauthlib: 1.3.1
        - requests-toolbelt: 1.0.0
        - rfc3339-validator: 0.1.4
        - rfc3986-validator: 0.1.1
        - rich:              13.7.0
        - rpds-py:           0.13.2
        - rsa:               4.9
        - s3transfer:        0.7.0
        - sacremoses:        0.0.53
        - safetensors:       0.3.3
        - scikit-image:      0.19.3
        - scikit-learn:      1.2.0
        - scipy:             1.11.4
        - seaborn:           0.12.2
        - secretstorage:     3.3.3
        - send2trash:        1.8.2
        - sentencepiece:     0.1.99
        - sentry-sdk:        1.32.0
        - setproctitle:      1.3.3
        - setuptools:        68.2.2
        - shellingham:       1.5.4
        - simpleitk:         2.2.1
        - sip:               6.7.12
        - six:               1.16.0
        - sklearn:           0.0.post1
        - smmap:             5.0.1
        - sniffio:           1.3.0
        - sortedcontainers:  2.4.0
        - soupsieve:         2.5
        - stack-data:        0.6.2
        - starlette:         0.27.0
        - starsessions:      1.3.0
        - sympy:             1.12
        - tables:            3.8.0
        - tabulate:          0.9.0
        - tblib:             3.0.0
        - tensorboard:       2.14.1
        - tensorboard-data-server: 0.7.1
        - tensorboardx:      2.6.2.2
        - terminado:         0.18.0
        - threadpoolctl:     3.1.0
        - tifffile:          2022.10.10
        - tinycss2:          1.2.1
        - tokenizer:         3.4.3
        - tokenizers:        0.13.3
        - tomli:             2.0.1
        - tomlkit:           0.12.3
        - toolz:             0.12.0
        - torch:             1.13.1
        - torchaudio:        0.13.1
        - torchmetrics:      1.2.1
        - torchsummary:      1.5.1
        - torchvision:       0.14.1
        - tornado:           6.3.3
        - tqdm:              4.64.1
        - traceback2:        1.4.0
        - traitlets:         5.14.0
        - traittypes:        0.2.1
        - transformers:      4.27.4
        - trove-classifiers: 2023.11.29
        - types-python-dateutil: 2.8.19.14
        - typeshed-client:   2.4.0
        - typing-extensions: 4.9.0
        - typing-utils:      0.1.0
        - tzdata:            2023.3
        - umap-learn:        0.5.4
        - unicodedata2:      15.1.0
        - unittest2:         1.1.0
        - uri-template:      1.3.0
        - urllib3:           2.1.0
        - uvicorn:           0.24.0.post1
        - vaex-astro:        0.9.3
        - vaex-core:         4.16.1
        - vaex-hdf5:         0.14.1
        - vaex-jupyter:      0.8.1
        - vaex-ml:           0.18.1
        - vaex-server:       0.8.1
        - vaex-viz:          0.5.4
        - virtualenv:        20.25.0
        - wandb:             0.16.1
        - wcwidth:           0.2.12
        - webcolors:         1.13
        - webencodings:      0.5.1
        - websocket-client:  1.7.0
        - websockets:        12.0
        - werkzeug:          3.0.0
        - wheel:             0.42.0
        - widgetsnbextension: 4.0.9
        - xarray:            2023.12.0
        - xxhash:            3.3.0
        - xyzservices:       2023.10.0
        - y-py:              0.5.9
        - yarl:              1.9.2
        - ypy-websocket:     0.8.2
        - zict:              3.0.0
        - zipp:              3.17.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.10.13
        - release:           5.4.0-167-generic
        - version:           #184-Ubuntu SMP Tue Oct 31 09:21:49 UTC 2023

</details>
